### PR TITLE
PEP 639: Post-split edits (language simplification, deeper edits of sections)

### PR DESCRIPTION
**Pull request by befeleme**
To be rebased on #3727 once merged.
It's best to review section by section rather than by commits.

### CODE REVIEW

**Comment by hugovk** [comment]
@befeleme Thanks for your work here, remember to add yourself as a co-author. Is @sethmlarson a co-author too, or just advising?

Let's also remove the `Content-Type: text/x-rst` header, it's no longer needed (all files have `.rst` extensions now).

**Review by ofek — APPROVED** [review]
Looks good to me, the Hatchling implementation matches what is written currently.

File: peps/pep-0639/appendix-license-survey.rst [hunk 1]
```diff
@@ -10,6 +10,23 @@ There are multiple ways used or recommended to document licenses.
 This document contains the results of a comprehensive survey of license
 documentation in Python and other languages.
 
+The key takeaways from the survey, which have guided the recommendations of
+PEP 639 are as follows:
```
Comment (JelleZijlstra) [inline peps/pep-0639/appendix-license-survey.rst:14]:
```suggestion
PEP 639, are as follows:
```

File: peps/pep-0639.rst [hunk 2]
```diff
@@ -18,97 +17,67 @@ Post-History: `15-Aug-2019 <https://discuss.python.org/t/2154>`__,
...
 
-The PEP also:
+To achieve that, it:
 
-- :ref:`Formally specifies <639-spec-field-license-file>`
-  a new ``License-File`` field, and defines how license files should be
-  :ref:`included in distributions <639-spec-project-formats>`,
-  as already used by the Wheel and Setuptools projects.
+- Adopts the `SPDX license expression syntax <639-spdx_>`__ as
+  means of expressing Python projects' license.
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:26]:
```suggestion
- Adopts the `SPDX license expression syntax <639-spdx_>`__ as a
  means of expressing the license for a Python project.
```

File: peps/pep-0639.rst [hunk 3]
```diff
@@ -18,97 +17,67 @@ Post-History: `15-Aug-2019 <https://discuss.python.org/t/2154>`__,
...
-  tools converting legacy license metadata, adding license files and
-  validating license expressions.
+- Describes the necessary changes to related specifications,
+  namely the `source distribution (sdist) <sdistspec_>`__,
+  `built distribution (wheel) <wheelspec_>`__ and
+  `installed project <installedspec_>`__ standards.
 
-- Describes a :ref:`reference implementation <639-reference-implementation>`
-  and analyzes numerous :ref:`potential alternatives <639-rejected-ideas>`.
+- :ref:`Provides clear guidance <639-spec-converting-metadata>`
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:39]:
```suggestion
- :ref:`Provides guidance <639-spec-converting-metadata>`
```
This word doesn't add any meaning here.

File: peps/pep-0639.rst [hunk 4]
```diff
@@ -18,97 +17,67 @@ Post-History: `15-Aug-2019 <https://discuss.python.org/t/2154>`__,
...
+If projects decide not to use the new fields, no additional restrictions are
+imposed by this PEP when uploading to PyPI.
 
-This PEP also is not about license documentation in files inside projects,
+This PEP also is not about license documentation for individual files,
 though this is a :ref:`surveyed topic <639-license-doc-source-files>`
-in an appendix, and nor does it intend to cover cases where the source and
-binary distribution packages don't have :ref:`the same licenses
-<639-rejected-ideas-difference-license-source-binary>`.
+in an appendix, and nor does it intend to cover cases where the
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:78]:
```suggestion
in an appendix, nor does it intend to cover cases where the
```

File: peps/pep-0639.rst [hunk 5]
```diff
@@ -117,317 +86,395 @@ Motivation
...
 `ClearlyDefined project <clearlydefined_>`__, an
 `Open Source Initiative <osi_>`__ incubated effort to help
 improve licensing clarity of other FOSS projects, covering all packages
 from PyPI, Maven, npm and Rubygems.
 
+The current license classifiers could be extended to include the full range of
+the SPDX identifiers while deprecating the ambiguous classifiers
+(such as ``License :: OSI Approved :: BSD License``).
+
+However, there are multiple reasons against such approach:
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:115]:
```suggestion
However, there are multiple arguments against such an approach:
```

File: peps/pep-0639.rst [hunk 6]
```diff
@@ -117,317 +86,395 @@ Motivation
...
+        originally introduced in :pep:`427`
+        and currently defined by
+        the PyPA spec :ref:`packaging:binary-distribution-format`.
+        Not to be confused with its reference implementation,
+        the :term:`Wheel project`.
+
+    Wheel project
+        The PyPA reference implementation of the :term:`Wheel format`.
+        To avoid ambiguity with the latter,
+        Will be consistently referred to as such here.
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:245]:
```suggestion
        the project will be consistently referred to as such here.
```

File: peps/pep-0639.rst [hunk 7]
```diff
@@ -117,317 +86,395 @@ Motivation
...
-
-**Wheel** *(Short: wheel, Rel: wheel format, Wheel project)*
-  Here, **wheel**, the standard built distribution format introduced in
-  :pep:`427` and `specified by the PyPA <wheelspec_>`__, will be referred to in
-  lowercase, while the `Wheel project <wheelproject_>`__, its reference
-  implementation, will be referred to as such with **Wheel** in Title Case.
+
+.. _639-terminology-packaging:
+
+Packaging terms
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:188]: This section feels like it should be a standalone PR to the packaging spec directly.

File: peps/pep-0639.rst [hunk 8]
```diff
@@ -117,317 +86,395 @@ Motivation
...
 .. _639-spec-core-metadata:
 
-Core metadata
+Core Metadata
 -------------
 
-The `PyPA Core Metadata specification <coremetadataspec_>`__ defines the names
-and semantics of each of the supported fields in the distribution metadata of
-Python distribution packages and installed projects.
+This PEP specifies following changes to the
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:443]:
```suggestion
This PEP specifies the following changes to the
```

File: peps/pep-0639.rst [hunk 9]
```diff
@@ -685,8 +753,50 @@ they create are legally distributable, build tools SHOULD default to
...
+    [project]
+    license-files = { globs = [] }
+
+Examples of invalid license files declaration:
+
+.. code-block:: toml
+
+    [project]
+    license-files.globs = ["LICEN[CS]E*", "AUTHORS*"]
+    license-file.paths = ["LICENSE.MIT"]
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:778]:
```suggestion
    license-files.paths = ["LICENSE.MIT"]
```
Also, it's confusing that none of the "valid" examples above use the dotted syntax. That made me think that there was something invalid about the dotted syntax.

File: peps/pep-0639.rst [hunk 10]
```diff
@@ -685,8 +753,50 @@ they create are legally distributable, build tools SHOULD default to
...
+Reason: license-files.paths and license-files.globs are mutually exclusive.
 
-.. _639-spec-key-license:
+.. code-block:: toml
+
+    [project]
+    license-files = { paths = ["..\LICENSE.MIT"] }
+
+Reason: ``..`` must not be used.
+``\`` is invalid path delimiter, ``/`` must be used.
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:788]:
```suggestion
``\`` is an invalid path delimiter, ``/`` must be used.
```

File: peps/pep-0639.rst [hunk 11]
```diff
@@ -792,72 +900,50 @@ the PEP authors.
...
+``pyproject.toml`` will only have an effect once users and tools adopt it.
+
+This PEP specifies that license files should be placed in a dedicated
+``licenses`` subdir of ``.dist-info`` directory. This is new and ensures that
+wheels following this PEP will have differently-located licenses relative to
+those produced via the previous installer-specific behavior. This is further
+supported by a new metadata version.
+
+This also resolves current issues where license files are accidentally
+replaced if they have the same names in different places, making wheel
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:927]:
```suggestion
replaced if they have the same names in different places, making wheels
```

File: peps/pep-0639.rst [hunk 12]
```diff
@@ -949,96 +1028,69 @@ Open Issues
...
-determined to be unnecessary or too problematic in practice.
+to handle the cases where projects have a license, but there is not a
+recognized SPDX license identifier for it. For maximum flexibility, custom
+``LicenseRef-<CUSTOM-TEXT>`` license identifiers could be allowed. In some cases
+``LicenseRef-Proprietary`` may not be appropriate or specific enough, but
+package authors could still want to benefit from the mainstream Python build
+tooling.
+
+However, this could increase the confusion about licensing. Custom identifiers
+can not be checked for correctness and users may think they always have to
```
Comment (JelleZijlstra) [inline peps/pep-0639.rst:1062]:
```suggestion
cannot be checked for correctness and users may think they always have to
```

File: peps/pep-0639.rst (hunk 9 above)
Comment (befeleme) [reply peps/pep-0639.rst:778]: Good point, will rework that.

File: peps/pep-0639.rst (hunk 7 above)
Comment (befeleme) [reply peps/pep-0639.rst:188]: I had similar feeling. Could we do that, @CAM-Gerlach?

**Comment by befeleme** [comment]
I went ahead and created https://github.com/pypa/packaging.python.org/pull/1541 with the glossary update, so that the section in PEP can be made thinner.

File: peps/pep-0639.rst (hunk 7 above)
Comment (befeleme) [reply peps/pep-0639.rst]: Done in https://github.com/pypa/packaging.python.org/pull/1541; references changed in c3c6e446381667b68d777404a093f75d49f3e3d2

**Comment by befeleme** [comment]
> @befeleme Thanks for your work here, remember to add yourself as a co-author. Is @sethmlarson a co-author too, or just advising?
> 
> Let's also remove the `Content-Type: text/x-rst` header, it's no longer needed (all files have `.rst` extensions now).

Both done now; @sethmlarson has advised.

At this point, this PEP is 5,138 words (14,808 including appendices) total. That's around 1/5 size shrink comparing to the state before this PR (18,238).
At this point, I'm starting getting blinded by the familiarity with the draft. I'd say this is the final state for the current edits.

What else is needed for this to be merged?

<details><summary>Details</summary>
<p>

```
wc -lw peps/pep-0639.rst peps/pep-0639/*             
  1013   5136 peps/pep-0639.rst
   217    662 peps/pep-0639/appendix-examples.rst
   380   2090 peps/pep-0639/appendix-license-survey.rst
   105    741 peps/pep-0639/appendix-mapping-classifiers.rst
   769   5274 peps/pep-0639/appendix-rejected-ideas.rst
   149    905 peps/pep-0639/appendix-user-scenarios.rst
  2633  14808 total
```

</p>
</details>

**Comment by hugovk** [comment]
Thanks!

Let's merge, then when you open the round 3 discussion (re: https://discuss.python.org/t/pep-639-round-2-improving-license-clarity-with-better-package-metadata/12622/113), people can read the published one.

After opening the discussion, please open a quick PR to point Discussions-To to it, and also add it to Post-History.
